File: CoreAutomata/IPattern.cs
﻿using System;
using System.Collections.Generic;

namespace CoreAutomata
{
    public interface IPattern : IDisposable
    {
        IPattern Resize(Size Size);

        void Resize(IPattern Target, Size Size);

        bool IsMatch(IPattern Template, double Similarity);

        IEnumerable<Match> FindMatches(IPattern Template, double Similarity);

        int Width { get; }

        int Height { get; }

        IPattern Crop(Region Region);

        void Save(string Filename);

        IPattern Copy();
    }
}

File: app/src/main/java/com/mathewsachin/fategrandautomata/core/IPattern.kt
package com.mathewsachin.fategrandautomata.core

interface IPattern : AutoCloseable {
    fun resize(Size: Size): IPattern

    fun resize(Target: IPattern, Size: Size)

    fun isMatch(Template: IPattern, Similarity: Double): Boolean

    fun findMatches(Template: IPattern, Similarity: Double): Sequence<Match>

    val width: Int
    val height: Int

    fun crop(Region: Region): IPattern

    fun save(FileName: String)

    fun copy(): IPattern
}